src/txreconciliation.cpp

```diff
@@ -0,0 +1,996 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <txreconciliation.h>
+
+#include <minisketch/include/minisketch.h>
+
+#include <unordered_map>
+
+namespace {
+
+/** Current protocol version */
+constexpr uint32_t RECON_VERSION = 1;
+/** Static component of the salt used to compute short txids for inclusion in sketches. */
+const std::string RECON_STATIC_SALT = "Tx Relay Salting";
+/** The size of the field, used to compute sketches to reconcile transactions (see BIP-330). */
+constexpr unsigned int RECON_FIELD_SIZE = 32;
+/**
+ * Allows to infer capacity of a reconciliation sketch based on it's char[] representation,
+ * which is necessary to deserealize a received sketch.
+ */
+constexpr unsigned int BYTES_PER_SKETCH_CAPACITY = RECON_FIELD_SIZE / 8;
+/**
+ * Limit sketch capacity to avoid DoS. This applies only to the original sketches,
+ * and implies that extended sketches could be at most twice the size.
+ */
+constexpr uint32_t MAX_SKETCH_CAPACITY = 2 << 12;
+/**
+* It is possible that if sketch encodes more elements than the capacity, or
+* if it is constructed of random bytes, sketch decoding may "succeed",
+* but the result will be nonsense (false-positive decoding).
+* Given this coef, a false positive probability will be of 1 in 2**coef.
+*/
+constexpr unsigned int RECON_FALSE_POSITIVE_COEF = 16;
+static_assert(RECON_FALSE_POSITIVE_COEF <= 256,
+    "Reducing reconciliation false positives beyond 1 in 2**256 is not supported");
+/** Default value for the coefficient used to estimate reconciliation set differences. */
+constexpr double DEFAULT_RECON_Q = 0.02;
```

Comment:
Interesting, default value sounds to diverge from Erlay paper recommendation : 

"The coefficient q characterizes earlier reconciliation, so before the very first reconciliation round it is set to zero."

You find empirically that a small coefficient was better to start with?

Reply:
For future conversations, I will refer to my findings [here](https://github.com/naumenkogs/txrelaysim/tree/master/results/flood-erlay).